src/algorithm/construct/LargestEmptyCircle.cpp

```diff
@@ -231,12 +232,18 @@ LargestEmptyCircle::compute()
      * Carry out the branch-and-bound search
      * of the cell space
      */
-    while (!cellQueue.empty()) {
+    std::size_t maxIter = MaximumInscribedCircle::computeMaximumIterations(boundary.get(), tolerance);
+    std::size_t iterationCount = 0;
+    while (!cellQueue.empty() && iterationCount < maxIter) {
```

Comment:
Should any action be taken if `maxIter` is hit while the queue is non-empty?

Reply:
For most practical cases the iteration converges very fast to a good approximation to the LEC centre.  The only other option I can see is to throw an exception warning that the result could be inaccurate.

Reply:
It seems like without an exception, you don't know if the result is correct or not?

Reply:
The algorithm is a heuristic approximation, so it will almost never find a perfectly correct answer anyway.  If the max iterations are exceeded, the result should be "close".  

I think there's an improvement that could be made to ensure that the computed point always lies inside the boundary/obstacle convex hull.  That fulfills the most important part of the contract.